### PR TITLE
Return shares for prover regardless of canon block status

### PR DESCRIPTION
**Pull request by jules**

### CODE REVIEW

File: storage/src/state/operator.rs [hunk 1]
```diff
@@ -129,10 +129,7 @@ impl<N: Network> SharesState<N> {
                 if !shares.contains_key(prover) {
                     None
                 } else {
-                    match ledger.contains_commitment(&coinbase_record.commitment()) {
-                        Ok(true) => shares.get(prover).copied(),
-                        Ok(false) | Err(_) => None,
-                    }
+                    shares.get(prover).copied()
```
Comment (ljedrz) [inline storage/src/state/operator.rs]: this doubles the lookup, no? instead, we should just do `.filter_map(|((_, coinbase_record), shares)| shares.get(prover).copied())`

Comment (jules) [reply storage/src/state/operator.rs]: completely right - just updated it

**Review by ljedrz — APPROVED** [review]
LGTM 👍

**Review by niklaslong — APPROVED** [review]

**Review by howardwu — APPROVED** [review]

**Comment by howardwu** [comment]
Great catch @julesdesmit

**Pull request merged**